### DISCUSSION TITLE
Hackathon: 2024-04-16

### DISCUSSION THREAD

**Discussion by stefaniebutland** [earthaccess-dev/earthaccess#522] [Hack days]
## Reporting out on earthaccess hack days.

Please create a new comment (at the very bottom, use the text box with a button that says "comment", not "reply") to write about what you did today!

Please include links to GitHub issues or pull requests relevant to your work, and use those issues or pull requests as the "source of truth" for documenting the work; this makes future readers jobs easier :) It's OK to duplicate some information while writing a summary here!

Use the "reply" feature to have a discussion under any given comment.

**Comment by mfisher87** [comment]
Myself, @briannapagan, @arieknoester, @doug-newman-nasa, and @andypbarrett worked on docs a bit together, then split in to two groups:

* Documentation structure (#498)
* Contributor documentation user testing (#524)
    * We also had a conversation about the strictness of the language in this doc. We feel the requirement to include unit tests in a change may be putting off some contributors. (#525)

**Reply by asteiker** [reply]
@MattF-NSIDC @andypbarrett I missed the conversation on #498 . Did you talk at all about #514 and whether to incorporate the latter into the larger restructuring? I like your suggestion to embed "Work with us" under @andypbarrett 's new Welcome section.

**Reply by mfisher87** [reply]
We talked about breaking the "Welcome" section up a bit and including the hack day content, and I think we're all positive on that!

I think I should probably rebase #514 onto #498. What do you all think?

**Reply by asteiker** [reply]
I think that makes the most sense. Sound good to you @andypbarrett ?

**Comment by andypbarrett** [comment]
Reconfigured Welcome page on #498 to break out Getting Started and Work with Us sections onto separate pages

Will push changes to PR

Plan to discuss documentation structure at next hackday to figure out what needs to be done so that PR can be merged to facilitate further community development of docs.

**Reply by andypbarrett** [reply]
Pushed changes to https://github.com/nsidc/earthaccess/pull/498

**Comment by doug-newman-nasa** [comment]
I had some notes that I forgot to convey in the meeting related to my last PR:

- When to do a `poetry lock --no-update`
- running pre-commit
- integration test instructions

**Comment by asteiker** [comment]
@JessicaS11 @danielfromearth @jhkennedy and I discussed alignment across [icepyx](https://icepyx.readthedocs.io/) ,`earthaccess` and [Harmony](https://github.com/nasa/harmony). `icepyx` currently supports on-prem transformation services provided by NSIDC DAAC and we need to start coordinating the transition to Harmony services. We discussed the role of NASA-provided services in the context of the cloud, Harmony's support of community development and community-provided backend services, how SlideRule fits into the transformation service landscape, and how we can provide guidance to users as they transition from the on-prem subsetter to Harmony. Ultimately we will continue these conversations especially in the context of #328 to enable Harmony services to be invoked from this library.
